### PR TITLE
Adds batch status to batch pages

### PR DESCRIPTION
**Pull request by maxkadel**
- Status not currently represented in the DB so not sortable
- Tried just adding a column, but needs more work wrt when to update status before that's ready

### CODE REVIEW

File: app/models/batch_process.rb [hunk 1]
```diff
@@ -89,4 +89,38 @@ def refresh_metadata_cloud
...
+    elsif statuses[:failed] != 0
+      percent = ((statuses[:failed] / statuses[:total]) * 100).round
+      "#{percent}% of parent objects have a failure."
+    else
+      "Batch status unknown"
+    end
+  end
+
+  def single_status
+    if statuses[:in_progress] / statuses[:total] == 1
```
Comment (mark-dce) [inline app/models/batch_process.rb:105]: I think you might want to capture the statuses one time here - I think the way you have this structured, you might be calling batch_connections.map up to 6 times.

```
current_status = statuses
if current_status[:in_progress] / current_status[:total] == 1
... etc.
```

File: app/models/batch_process.rb [hunk 2]
```diff
@@ -89,4 +89,38 @@ def refresh_metadata_cloud
...
+      "Batch in progress - no failures"
+    elsif statuses[:complete] / statuses[:total] == 1
+      "Batch complete"
+    elsif statuses[:failed] / statuses[:total] == 1
+      "Batch failed"
+    end
```
Comment (mark-dce) [inline app/models/batch_process.rb:111]: Let's add an else here to catch any unexpected states

```
else
    "Unexpected state"

**Review by mlooney — APPROVED** [review]

**Review by mark-dce — CHANGES_REQUESTED** [review]
This looks great.  I've marked two small changes in calculating status that I think might make the status calculation faster and catch a possible future error state that we "shouldn't " be able to reach, but might at some point in the future if someone refactors statuses.

File: app/models/batch_process.rb [hunk 3]
```diff
@@ -89,4 +89,38 @@ def refresh_metadata_cloud
...
+  end
+
+  def statuses
+    connected_statuses = batch_connections.map do |batch_connection|
```
Comment (orangewolf) [inline app/models/batch_process.rb:115]: this method can be refactored for speed up, I think significantly.

**Pull request merged**